src/Interpreters/sortBlock.cpp

```diff
@@ -194,27 +189,16 @@ void getBlockSortPermutationImpl(const Block & block, const SortDescription & de
 
 void sortBlock(Block & block, const SortDescription & description, UInt64 limit)
 {
-    IColumn::Permutation permutation;
-    getBlockSortPermutationImpl(block, description, IColumn::PermutationSortStability::Unstable, limit, permutation);
-
-    if (permutation.empty())
-        return;
-
-    size_t columns = block.columns();
-    for (size_t i = 0; i < columns; ++i)
+    if (isAlreadySorted(block, description))
     {
-        auto & column_to_sort = block.getByPosition(i).column;
-        column_to_sort = column_to_sort->permute(permutation, limit);
-    }
-}
-
-void stableSortBlock(Block & block, const SortDescription & description)
-{
-    if (!block)
+        if (limit && block.rows() > limit)
+            for (auto & elem : block)
+                elem.column = elem.column->cut(0, limit);
         return;
+    }
 
     IColumn::Permutation permutation;
-    getBlockSortPermutationImpl(block, description, IColumn::PermutationSortStability::Stable, 0, permutation);
+    getBlockSortPermutationImpl(block, description, IColumn::PermutationSortStability::Unstable, limit, permutation);
```

Comment:
`getBlockSortPermutationImpl()` can be made `static` since used only in this file

Reply:
It was already in the anonymous namespace.